receipt_listener.go

```diff
@@ -71,66 +73,58 @@ func (l *ReceiptListener) Run(ctx context.Context) error {
 			return nil
 
 		case blocks := <-sub.Blocks():
-			block := blocks.LatestBlock().Block
-			if block == nil {
-				l.log.Warn().Msgf("monitor return latestblock of nil, unexpected but skipping..")
-				continue
-			}
-
-			err := br.Do(ctx, func() error {
-				return l.handleBlock(ctx, block)
-			})
-
-			if err != nil {
-				if errors.Is(err, breaker.ErrHitMaxRetries) {
-					l.log.Err(err).Msgf("failed to handle block %d after many retries", block.NumberU64())
-					continue
-				} else {
-					l.log.Err(err).Msgf("failed to handle block %d", block.NumberU64())
-					continue
+			for _, block := range blocks {
+				if block.Event == ethmonitor.Added {
```

Comment:
I think its better to do like.. 

```go
if block.Event != ethmonitor.Added {
  continue
}
```

the reason it creates less nesting so its easier to read. its very common practice in go to short-circuit a method sooner or continue to next iteration within the block as a sort of assertion